### PR TITLE
Feature/separate page for partners

### PR DESCRIPTION
**Pull request by Camorte**

### CODE REVIEW

**Comment by luk-str** [comment]
I just generated all the partner slugs in sanity (thought we did that before but they were missing 🤷 ).

You can preview the partner sites by going to url/partner/slug so e.g. [https://deploy-preview-221--oslopride-web.netlify.app/partner/nordic-choice-hotels](https://deploy-preview-221--oslopride-web.netlify.app/partner/nordic-choice-hotels) should work 👍

**Comment by boyeborg** [comment]
@luk-str I remember thinking "this will cause some confusion some day" when I created the different sanity datasets. I guess "someday" is "today". Anyway. The way it works is that there are two main sanity datasets: one for production called `future` (for some historical reason), and one for the preview URLs called `staging`. Which dataset thats being used is configured through [the netlify.toml file](https://github.com/oslopride/oslopride.no/blob/c464c734389a10f3f6a5f1650ee01c38a5a61ea8/netlify.toml#L2).

If it would be possible (maybe it is now, but it wasn't when I created this workflow) it would be best if a copy of the production dataset would be created for each pull request's preview. However, since that isn't (wasn't?) possible, I settled on creation _one_ dataset for _all_ the previews. The reasoning behind this was that we probably wouldn't have too many PRs open at the same time that potentially could ruin each others dataset. Since this one preview dataset would be the one used for all previews, I figured that it would probably be best it it was somewhat up to date with production, so I decided that it should erase it selv, and reset to mirror production (future) every night (03:00 UTC). This is done through [a github action](https://github.com/oslopride/oslopride.no/blob/c464c734389a10f3f6a5f1650ee01c38a5a61ea8/.github/workflows/main.yml).

I think this system is due for some rethinking. First of all, its very inconvenient that different pull request previews work on the same dataset. Especially if they are altering something in sanity the data structure. Secondly, its super annoying that the preview dataset resets itself every night. For example with this preview, you made the appropriate changes to sanity yesterday, but today they're gone again. If anyone has any thoughts, please share!

**Comment by luk-str** [comment]
I just added some extra styling to make this work on mobile screens. Other than that I think this is basically ready to roll 👌

**Comment by Camorte** [comment]
> I just added some extra styling to make this work on mobile screens. Other than that I think this is basically ready to roll 👌

Ah, I forgot to add it on the issue, but we're still missing a new field in Sanity to add Content below the description and logo. After that I think we are good to go!

File: sanity/schemas/partner.js [hunk 1]
```diff
@@ -39,6 +39,30 @@ export default {
...
+		},
+		{
+			title: "Instagram social link",
+			name: "instagramLink",
+			type: "string"
+		},
+		{
+			title: "LinkedIn social link",
+			name: "linkedinLink",
+			type: "string"
```
Comment (hexjelly) [inline sanity/schemas/partner.js:65]: I don't know if this makes any difference (or if it helps in any way at all?), but I think there should be a Sanity type called `url`, which maybe makes more sense than just a plain `string` as the type for these? Not sure if it appears differently in Sanity or what the deal is though.

Comment (boyeborg) [reply sanity/schemas/partner.js:65]: I think the URL adds some validation logic that might be nice.

Comment (Camorte) [reply sanity/schemas/partner.js:65]: Agreed, I forgot that the url type existed. I'll be working on the rest of the task tomorrow!

**Review by boyeborg — CHANGES_REQUESTED** [review]
Looks nice! The only thing I believe is missing, is an article/body/content section that allows the marketing team to create a more interesting page for the different partners. So there should be a sanity field called body or article (or something like that), similar to the `Body` field in the articles. You can probably just copy/paste the same code:

```
  title: "Body",
  name: "body",
  type: "array",
  of: [
    { type: "block" },
    {
      type: "image",
      options: { hotspot: true }
    },
    { type: "youtube" },
    { type: "iframe" }
  ]
```

When rendering it (in react), just use the `sanity-portable-text` component. It has all the correct serializers.

If you're extra, you could also restrict the styles they are allowed to use in both the body and the description fields. Sanity [has a good article on how you configure the `block` type here](https://www.sanity.io/docs/configuration). I think it would probably make sense to only allow **bold** and _italic_ in the description, and perhaps allow everything except `h1` in the body? Or maybe rename the heading options so h2 is called "Heading" and "h3" is called Subheading, and only allow those?

File: web/src/pages/partner.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+									margin: 0;
+								`}
+							>
+								{partner.name}
+							</h2>
+							<BlockContentToReact blocks={partner.description} />
+
+							<div>
+								{partner.facebookLink && (
+									<FBSocialLink
```
Comment (boyeborg) [inline web/src/pages/partner.tsx:170]: The social links should perhaps be actual clickable links 🤪

Comment (Camorte) [reply web/src/pages/partner.tsx:170]: Hehe, fake it til you make it, no? Good catch :D

**Review by boyeborg — APPROVED** [review]
Looks great! Ship it! 🚢

**Pull request merged**